flang/lib/Optimizer/Builder/IntrinsicCall.cpp

```diff
@@ -6366,8 +6387,9 @@ IntrinsicLibrary::genLbound(mlir::Type resultType,
   // LBOUND(array, dim, [kind]) or LBOUND(array, [kind]).
   const bool dimIsAbsent = args.size() == 2 || isStaticallyAbsent(args, 1);
   if (array.hasAssumedRank() && dimIsAbsent)
-    return genAssumedRankBoundInquiry(builder, loc, resultType, args,
-                                      /*kindPos=*/1, fir::runtime::genLbound);
+    return genBoundInquiry(builder, loc, resultType, args,
+                           /*kindPos=*/1, fir::runtime::genLbound,
```

Comment:
Is the `kindPos` correct for the case of statically absent dim?

Reply:
No, I updated it to take the arg size into account like I did in Ubound. Thanks for catching this!